internal/helpers/schemaregistry/schema_registry.go

```diff
@@ -0,0 +1,82 @@
+package schemaregistry
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/protoparse"
+	"github.com/pkg/errors"
+	"github.com/riferrei/srclient"
+)
+
+// FormatBaseURL will try to make sure that the schema:host:port:path pattern is followed on the `baseURL` field.
+func FormatBaseURL(baseURL string) string {
+	if baseURL == "" {
+		return ""
+	}
+
+	// remove last slash, so the API can append the path with ease.
+	if baseURL[len(baseURL)-1] == '/' {
+		baseURL = baseURL[0 : len(baseURL)-1]
+	}
+
+	schemaIdx := strings.Index(baseURL, "://")
+	injectSchema := schemaIdx < 0
+
+	if injectSchema {
+		baseURL = fmt.Sprintf("http://%s", baseURL)
+	}
+
+	parsedUrl, err := url.Parse(baseURL)
+	if err != nil {
+		panic("Schema registry url invalid")
+	}
+
+	if injectSchema && parsedUrl.Port() == "443" {
+		parsedUrl.Scheme = "https"
+	}
+
+	if parsedUrl.Port() == "" {
+		if parsedUrl.Scheme == "https" {
+			parsedUrl.Host = fmt.Sprintf("%s:%s", parsedUrl.Hostname(), "443")
+		} else {
+			parsedUrl.Host = fmt.Sprintf("%s:%s", parsedUrl.Hostname(), "80")
+		}
+	}
+
+	return parsedUrl.String()
+}
+
+func SchemaToFileDescriptor(registry srclient.ISchemaRegistryClient, schema *srclient.Schema) (*desc.FileDescriptor, error) {
```

Comment:
this function and the two below are probably for a next PR? For this one they can be removed.

Reply:
Yes. Sure. I'll remove it. You're right. It's future change